### PR TITLE
fix(docker): publish container image upon successful release

### PR DESCRIPTION
**Pull request by dpopp07**
The container was not being published because it was set up to push on a tagged build, but we skip tagged builds (which I forgot about). This updates the image publishing to push from semantic release's "success" step, which will guarantee it gets pushed after the creation of a new release during the main branch build.

### CODE REVIEW

File: scripts/deploy-container-image.sh [hunk 1]
```diff
@@ -19,14 +19,12 @@ login() {
 
 
 deploy_docker() {
-    local tag=$1
-    [[ "$tag" =~ ibm-openapi-validator@([[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+) ]]
-    local docker_tag=${BASH_REMATCH[1]}
```
Comment (dpopp07) [inline scripts/deploy-container-image.sh:24]: It was easier to give the script the raw version (`x.x.x`), which also saved us the parsing step, so I went with that.

Also, I decided to rename the file to clarify what exactly was being deployed

File: scripts/deploy-container-image.sh [hunk 2]
```diff
@@ -19,14 +19,12 @@ login() {
...
+    local new_version=$1
```
[the remark this answers is not in the document]
Comment (dpopp07) [reply scripts/deploy-container-image.sh:21]: Good call. Just pushed a change to parse the semver string from the received string (which _should_ always be a semver string) and ensure it is not missing before pushing the image.

**Review by padamstx — APPROVED** [review]
Looks good!

**Pull request merged**